fido2/hid/freebsd.py

```diff
@@ -0,0 +1,62 @@
+# Original work Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Modified work Copyright 2020 Yubico AB. All Rights Reserved.
+# This file, with modifications, is licensed under the above Apache License.
+
+
+from __future__ import absolute_import
+
+import os
+import uhid_freebsd
+
+from .base import HidDescriptor, parse_report_descriptor, FileCtapHidConnection
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def open_connection(descriptor):
+    return FileCtapHidConnection(descriptor)
+
+
+def _read_descriptor(vid, pid, path):
+    fd = os.open(path, os.O_RDONLY)
+    data = uhid_freebsd.get_report_data(fd, 3)
```

Comment:
I believe this value is supposed to be this enum: https://docs.huihoo.com/doxygen/linux/kernel/3.7/uhid_8h_source.html#l00055

However, 3 doesn't look like it would be a valid number.
The uhid-freebsd project has an example in the README which also uses the constant 3, so for now I think we'll just leave it as-is.